src/charm.py

```diff
@@ -54,25 +55,27 @@ def __init__(self, *args):
         self.framework.observe(self.on[ZK].relation_changed, self._on_config_changed)
         self.framework.observe(self.on[ZK].relation_broken, self._on_zookeeper_broken)
 
-        self.framework.observe(self.on.set_password_action, self._set_password_action)
+        self.framework.observe(getattr(self.on, "set_password_action"), self._set_password_action)
```

Comment:
?

Reply:
Fixes type-hint warning, as `set_password_action` isn't a known attr of `self.on`

Reply:
Despite the type-hint warning I think the pervious version is more readable